src/coreclr/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -796,15 +796,127 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
+
+                            GenTree* addr = intrin.op1->AsIndir()->Addr();
+                            assert(!addr->isContained());
+                            baseReg = addr->GetRegNum();
+                        }
+                    }
+                    else
+                    {
+                        unsigned simdInitTempVarNum = compiler->lvaSIMDInitTempVarNum;
+                        noway_assert(simdInitTempVarNum != BAD_VAR_NUM);
+
+                        baseReg = node->ExtractTempReg();
+
+                        // Load the address of simdInitTempVarNum
+                        GetEmitter()->emitIns_R_S(INS_lea, EA_PTRSIZE, baseReg, simdInitTempVarNum, 0);
+
+                        // Store the vector to simdInitTempVarNum
+                        GetEmitter()->emitIns_R_R(INS_str, emitTypeSize(simdType), op1Reg, baseReg);
+                    }
+
+                    assert(genIsValidIntReg(indexReg));
+                    assert(genIsValidIntReg(baseReg));
+                    assert(baseReg != indexReg);
+
+                    // Load item at baseReg[index]
+                    GetEmitter()->emitIns_R_R_R_Ext(ins_Load(intrin.baseType), baseTypeSize, targetReg, baseReg,
+                                                    indexReg, INS_OPTS_LSL, baseTypeScale);
+                }
+                else if (!GetEmitter()->isValidVectorIndex(emitTypeSize(simdType), emitTypeSize(intrin.baseType),
+                                                           intrin.op2->AsIntCon()->IconValue()))
+                {
+                    // We only need to generate code for the get if the index is valid
+                    // If the index is invalid, previously generated for the range check will throw
+                }
+                else if (!intrin.op1->isUsedFromReg())
+                {
+                    assert(intrin.op1->isContained());
+                    assert(intrin.op2->IsCnsIntOrI());
+
+                    int         offset = (int)intrin.op2->AsIntCon()->IconValue() * genTypeSize(intrin.baseType);
+                    instruction ins    = ins_Load(intrin.baseType);
+
+                    assert(!intrin.op1->isUsedFromReg());
+
+                    if (intrin.op1->OperIsLocal())
+                    {
+                        unsigned varNum = intrin.op1->AsLclVarCommon()->GetLclNum();
+                        GetEmitter()->emitIns_R_S(ins, emitActualTypeSize(intrin.baseType), targetReg, varNum, offset);
+                    }
+                    else
+                    {
+                        assert(intrin.op1->OperIs(GT_IND));
+
+                        GenTree* addr = intrin.op1->AsIndir()->Addr();
+                        assert(!addr->isContained());
+                        regNumber baseReg = addr->GetRegNum();
+
+                        // ldr targetReg, [baseReg, #offset]
+                        GetEmitter()->emitIns_R_R_I(ins, emitActualTypeSize(intrin.baseType), targetReg, baseReg,
```

Comment:
What will we generate when `intrin.baseType` is `byte`/`sbyte`/`short`/`ushort`? 

I presume, `ins` should be `ldrb`/`ldrsb`/`ldrsh`/`ldrh`? What should be `sizeAttr` corresponding to these types? Should the attribute be computed by `emitTypeSize` or `emitActualTypeSize`? Do we have tests that covers these scenarios?

Reply:
We should via the template tests yes. This is really just copying what the Simd apth was already doing, so there shouldn't be any new logic